src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke.storage;
+
+import duke.exception.DukeException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedList;
+
+public class Storage {
+
+    private String filepath;
+
+    /**
+     * Initialises a Storage session.
+     *
+     * @param filepath The file to store tasks to.
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Reads the input file line by line to add tasks into the TaskList.
+     *
+     * @return The LinkedList containing all loaded tasks.
+     * @throws DukeException In the event that the file cannot be read properly.
+     */
+    public LinkedList<Task> load() throws DukeException {
+        LinkedList<Task> allTasks = new LinkedList<>();
```

Comment:
Usually, it is recommended to use `ArrayList` instead of `LinkedList` for faster access of `Tasks`.